.drone.yml

```diff
@@ -113,8 +94,16 @@ steps:
       ssh_key:
         from_secret: github_private_key
       commit: true
-      tag: v$${DRONE_BUILD_NUMBER}
-      followtags: true     
+      tag: v${DRONE_BUILD_NUMBER}
+      followtags: true
+    when:
+      event:
+       - push
+      branch:
+       - main  
+    depends_on:
+      - install dependencies
```

Comment:
I don't think you need to depend on 'install dependencies' as 'sonar analysis' depends on 'test project' which in turn depends on 'install dependencies'.